### PR TITLE
Introducing Oracle Free module

### PR DESCRIPTION
**Pull request by gvenzl**
This PR introduces a new Oracle Database Free module, the successor to Oracle XE

### CODE REVIEW

File: modules/oracle-free/src/main/java/org/testcontainers/containers/OracleContainer.java [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+    private boolean usingSid = false;
+
+    /**
+     * @deprecated use {@link #OracleContainer(DockerImageName)} instead
+     */
+    @Deprecated
+    public OracleContainer() {
+        this(DEFAULT_IMAGE_NAME.withTag(DEFAULT_TAG));
+    }
```
Comment (eddumelendez) [inline modules/oracle-free/src/main/java/org/testcontainers/containers/OracleContainer.java:70]: Let's remove this constructor

File: modules/oracle-free/src/main/java/org/testcontainers/containers/OracleContainer.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+/**
+ * Testcontainers implementation for Oracle Database Free.
+ * <p>
+ * Supported image: {@code gvenzl/oracle-free}
+ * <p>
+ * Exposed ports: 1521
+ */
+public class OracleContainer extends JdbcDatabaseContainer<OracleContainer> {
+
+    public static final String NAME = "oracle";
```
Comment (eddumelendez) [inline modules/oracle-free/src/main/java/org/testcontainers/containers/OracleContainer.java:25]:
```suggestion
    static final String NAME = "oracle";
```

File: modules/oracle-free/src/main/java/org/testcontainers/containers/OracleContainer.java [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    private void preconfigure() {
+        this.waitStrategy =
+            new LogMessageWaitStrategy()
+                .withRegEx(".*DATABASE IS READY TO USE!.*\\s")
+                .withTimes(1)
+                .withStartupTimeout(Duration.of(DEFAULT_STARTUP_TIMEOUT_SECONDS, ChronoUnit.SECONDS));
+
+        withConnectTimeoutSeconds(DEFAULT_CONNECT_TIMEOUT_SECONDS);
+        addExposedPorts(ORACLE_PORT);
+    }
```
Comment (eddumelendez) [inline modules/oracle-free/src/main/java/org/testcontainers/containers/OracleContainer.java:96]:
```suggestion
        this.waitStrategy =
            new LogMessageWaitStrategy()
                .withRegEx(".*DATABASE IS READY TO USE!.*\\s")
                .withTimes(1)
                .withStartupTimeout(Duration.of(DEFAULT_STARTUP_TIMEOUT_SECONDS, ChronoUnit.SECONDS));

        withConnectTimeoutSeconds(DEFAULT_CONNECT_TIMEOUT_SECONDS);
        addExposedPorts(ORACLE_PORT);
    }
```

File: modules/oracle-free/src/main/java/org/testcontainers/containers/OracleContainer.java [hunk 4]
```diff
@@ -0,0 +1,212 @@
+package org.testcontainers.containers;
```
Comment (eddumelendez) [inline modules/oracle-free/src/main/java/org/testcontainers/containers/OracleContainer.java:1]:
```suggestion
package org.testcontainers.oracle;
```

**Review by eddumelendez — COMMENTED** [review]
Thanks for your contributions, @gvenzl ! I've left some comments and it would be great to add some docs in `mkdocs.yml`and `docs/modules/databases`. Also, update the following files `bug_report.yaml`, `enhancement.yaml`, `feature.yaml`, `dependabot.yml` and `labeler.yml`.

File: modules/oracle-free/src/main/java/org/testcontainers/containers/OracleContainer.java (hunk 1 above)
Comment (gvenzl) [reply modules/oracle-free/src/main/java/org/testcontainers/containers/OracleContainer.java]: Done

File: modules/oracle-free/src/main/java/org/testcontainers/containers/OracleContainer.java [hunk 5]
```diff
@@ -0,0 +1,200 @@
+package org.testcontainers.oracle;
```
Comment (eddumelendez) [inline modules/oracle-free/src/main/java/org/testcontainers/containers/OracleContainer.java:1]: the package has not been renamed. It is still `org.testcontainers.containers`

File: modules/oracle-free/src/main/java/org/testcontainers/containers/OracleContainer.java [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+        super(dockerImageName);
+        this.waitStrategy =
+            new LogMessageWaitStrategy()
+                .withRegEx(".*DATABASE IS READY TO USE!.*\\s")
+                .withTimes(1)
+                .withStartupTimeout(Duration.of(DEFAULT_STARTUP_TIMEOUT_SECONDS, ChronoUnit.SECONDS));
+
+        withConnectTimeoutSeconds(DEFAULT_CONNECT_TIMEOUT_SECONDS);
+        addExposedPorts(ORACLE_PORT);
+    }
```
Comment (eddumelendez) [inline modules/oracle-free/src/main/java/org/testcontainers/containers/OracleContainer.java:84]:
```suggestion
    public OracleContainer(String dockerImageName) {
        this(DockerImageName.parse(dockerImageName));
    }

    public OracleContainer(final DockerImageName dockerImageName) {
        super(dockerImageName);
        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);

        this.waitStrategy =
            new LogMessageWaitStrategy()
                .withRegEx(".*DATABASE IS READY TO USE!.*\\s")
                .withTimes(1)
                .withStartupTimeout(Duration.of(DEFAULT_STARTUP_TIMEOUT_SECONDS, ChronoUnit.SECONDS));

        withConnectTimeoutSeconds(DEFAULT_CONNECT_TIMEOUT_SECONDS);
        addExposedPorts(ORACLE_PORT);
    }
```

Comment (gvenzl) [reply modules/oracle-free/src/main/java/org/testcontainers/containers/OracleContainer.java]: Implemented

**Comment by eddumelendez** [comment]
Thanks again for your contribution, @gvenzl!

**Comment by gvenzl** [comment]
Any time! :)

File: modules/oracle-free/src/main/java/org/testcontainers/oracle/OracleContainer.java [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Testcontainers implementation for Oracle Database Free.
+ * <p>
+ * Supported image: {@code gvenzl/oracle-free}
```
Comment (ecki) [inline modules/oracle-free/src/main/java/org/testcontainers/oracle/OracleContainer.java:19]: Can you please also accept the official Oracle image.

File: modules/oracle-free/src/main/java/org/testcontainers/oracle/OracleContainer.java [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+    static final String DEFAULT_SID = "free";
+
+    static final String DEFAULT_SYSTEM_USER = "system";
+
+    static final String DEFAULT_SYS_USER = "sys";
+
+    // Test container defaults
+    static final String APP_USER = "test";
+
+    static final String APP_USER_PASSWORD = "test";
```
Comment (ecki) [inline modules/oracle-free/src/main/java/org/testcontainers/oracle/OracleContainer.java:51]: We really should default to secure random passwords in the future (in the baseclass)

**Pull request merged**